### ISSUE TITLE
Continue debugging when instrumented functions finishes executing and continue debugging caller

### ISSUE THREAD

**Issue by AhmedAzzabi** [clojure-emacs/cider-nrepl#516]
In cider for the debugger, if we have 2 functions func1 that calls func2 and only func2 is instrumented with #dbg, when evaluating func1 the debugger stops in func2 which is good, but after doing next and finishing all func2 is there a way to continue debugging the caller func1 even if it is not instrumented, currently it just finish the evaluation. I think when debugging it will be good to not have to manually instrument func1 or all the call hieararchy to continue debugging, i whish that as long as im doing next it just continue debugging step by step and when func2 finishes it steps out and continue debugging.

Thanks

**Comment by Malabarba** [comment]
No, there's no way to do that right now.
Unfortunately, due to how the debugger is implemented, such a feature would also be quite difficult to implement.

**Comment by vspinu** [comment]
> also im wondering if during debug i instrument the calling func1, will it take effect in current debug session running ? 

I think this is the right way to fix this issue.

**Comment by AhmedAzzabi** [comment]
Yes i think if we have the ability to add breakpoint while debug session is running we can solve this issue without much effort, we have stack trace and we can instrument calling functions, also it is a good feature to have anyway sometimes when im debugging i want to add more breakpoints but they are not taken in consideration and have to restart debug session in order for new breakpoints to take effect

**Comment by vspinu** [comment]
None of these is easy with the current implementation of the debugger. But if clojure-emacs/cider#2005 could be implemented it would essentially come for free. I have some ideas of how to debugg without hijacking the eval op but that would require a major rewrite of the debugger.
